packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1006,6 +1006,24 @@ class RawScrollbar extends StatefulWidget {
        assert(minOverscrollLength == null || minOverscrollLength >= 0),
        assert(radius == null || shape == null);
 
+  /// Creates a set of scrollbars for vertical and horizontal axes.
+  factory RawScrollbar.dual({
+    required ScrollController verticalController,
+    required ScrollController horizontalController,
+    required Widget child,
+  }) {
+    // TODO(Piinks): Instead, `.dual` should be a constructor that internally
+    //  creates two ScrollbarPainters so that they can respond to each other
+    //  appropriately: https://github.com/flutter/flutter/issues/122345
+    return RawScrollbar(
```

Comment:
We should override the outer `RawScrollbar.notificationPredicate` to let them work, right?

Maybe we can write a sample to show how to use this new feature : )

Reply:
We should also remove the default bar for desktop platforms if users choose the dual bars.

Reply:
That should not be necessary, since the new TwoDimensionalScrollable maintains the same notification depth for both axes.
Plus, it overrides the default behavior with this method instead of the other buildScrollbar. :)

See this for a rough example, finishing up some of the details this week: 
- https://github.com/flutter/flutter/pull/120622
- https://github.com/flutter/flutter/pull/120623

Reply:
Can we let this also support two nested scrollable widgets? Folks may not know this can only work for TwoDimensionalScrollable.

Just watched [this](https://www.youtube.com/watch?v=UDZ0LPQq-n8), it's amazing!

Reply:
`the new TwoDimensionalScrollable maintains the same notification depth for both axes.`

This may have an impact on the Scrollbar implementation because the scroll notification has the same depth but a different axis direction.

Reply:
> This may have an impact on the Scrollbar implementation because the scroll notification has the same depth but a different axis direction.

It actually works! I have been using it for months while I have been working on this. Just the corner space needs better handling (filed an issue linked above). And the scroll controllers must be assigned, there would be a lot of room for error if the scrollbar was just trying to respond blindly to notifications in this case. :)

Reply:
> Can we let this also support two nested scrollable widgets? Folks may not know this can only work for TwoDimensionalScrollable.

I think it is reasonable to expose the notification predicate in some way, but by default this dual composition should just listen to a depth of 1 as it is intended to be used with the 2D scrolling classes I am working on. Trying to self-compose 2D scrolling has a lot of issues (which is why we're working on making it easier!), and I don't want to overcomplicate a new feature by trying to account for every possible composition folks might have created themselves. I'll make sure this is documented really well, although I cannot reference TwoDimensionalScrollable until it lands (soon!).

Of course, we could get some feedback from users once this lands and decide to change it, that's why I love this project. 😊  Right now my goal is to get a solid basic foundation in place, and build from there.